4-js/index.js

```diff
@@ -0,0 +1,330 @@
+    button.innerText = i;
+    button.name = i * 10;
+    button.classList.add('paginator-number');
+    paginator.appendChild(button);
+    if (i === 1) {
+      button.classList.add('active');
+    }
+    button.addEventListener('click',showPagination);
+  }
+  getUsersInfo('_start=' + (quantityPagination * 10 - 10) + '&_limit=50');
+}
+
+function showPagination() {
+  getUsersInfo('_start=' + (paginations.length * 10 - this.name) + '&_limit=50');
+  changeStylePagination(this);
+}
+
+function changeStylePagination(el) {
+  const arrPaginations = document.getElementsByClassName('paginator-number');
+  for (let i = 0; i < arrPaginations.length; i++) {
+    arrPaginations[i].classList.remove('active');
+  }
+  el.classList.add('active');
+}
+
+function addInputListener() {
+    for (let i = 0; i < inputFilter.length; i++) {
+        inputFilter[i].addEventListener('keyup',e => {
+            if (e.keyCode >= 48&&e.keyCode <= 90||e.keyCode === 8) {
+                let resultSort = '';
+                if (companyInput.value !== '') {
+                    resultSort += 'company_like=' + companyInput.value + '&&';
+                }
+                if (nameInput.value !== '') {
+                    resultSort += 'name_like=' + nameInput.value + '&&';
+                }
+                if (addressInput.value !== '') {
+                    resultSort += 'address_like=' + addressInput.value + '&&';
+                }
+                if (e.keyCode === 8&&companyInput.value === ''&&nameInput.value === ''&&addressInput.value === '') {
+                  getUsersInfo('_start=' + (paginations.length * 10 - getCurrentNumberPogination()) + '&_limit=50');
+                } else {
+                  getUsersInfo(resultSort);
+                }
+            }
+        })
+    }
+}
+
+function getCurrentNumberPogination() {
+  let el = document.querySelector('.active');
+  return el.name;
+}
+
+function createUser() {
+  let newUser = {
+    company: userCompany.value,
+    name: userName.value,
+    address: userAddress.value,
+    city: userCity.value,
+    country: userCountry.value,
+  };
+  sendRequsest('POST', requestUrl, newUser)
+    .then((data) => {
+      createTableUsers([data],'dont-delete');
+    })
+    .catch((err) => console.log(err));
+}
+
+function formCheck() {
+  if (
+    userCompany.value !== '' &&
+    userName.value !== '' &&
+    userAddress.value !== '' &&
+    userCity.value !== '' &&
+    userCountry.value !== ''
+  ) {
+    createUserBtn.removeAttribute('disabled', 'disabled');
+    createUserBtn.classList.remove('is-translucent');
+  } else {
+    createUserBtn.classList.add('is-translucent');
+  }
+}
+
+function getUsersInfo(arg) {
+  sendRequsest('GET', requestUrl + '?' + arg)
+    .then((data) => createTableUsers(data))
+    .catch((err) => console.log(err));
+}
+
+function createTableUsers(arr, check) {
+  const tableUser = document.getElementsByClassName('table-user')[0];
+  if (check !== 'dont-delete') {
```

Comment:
what this condition for?

Reply:
When I add a new user, so that I don't have to query the server and redraw everything, I simply add the item to the top of the list.